### DISCUSSION TITLE
Option to disallow access to the user directory

### DISCUSSION THREAD

**Discussion by krazyjakee** [directus/directus#4608] [Feature Requests]
Quite simply, I don't want users to have access to all other users and their data.
Seeing the authors in the collections view should remain intact.

This got me close: https://github.com/directus/directus/discussions/3657 but the last piece is to be able to see content authors.

**Comment by rijkvanzanten** [comment]
nelolka's response is the recommended solution here: https://github.com/directus/directus/discussions/3657#discussioncomment-277894

Seeing you still want the users to have read access to other users (for user-created etc), you can't fully disable all read access to users. The next best thing is to remove the users tab from the sidebar.

**Reply by uebmaster** [reply]
The idea of remove the users tab is fine, but when the user enters to edit their profile they see all the user's groups :/

![image](https://user-images.githubusercontent.com/5565506/123452286-e13a1e00-d5a3-11eb-87fe-6380f1c24e68.png)

**Reply by benhaynes** [reply]
Not much we can do about that with making some pretty big changes (unless permissions can handle this). Any thoughts?

**Reply by rijkvanzanten** [reply]
The shown roles are based on the permissions of the user to directus_roles 🤔

**Reply by uebmaster** [reply]
As always Rijk has reason.

But initial question remains the same (and in my case too):

> Seeing the authors in the collections view should remain intact.

**Reply by uebmaster** [reply]
> Not much we can do about that with making some pretty big changes (unless permissions can handle this). Any thoughts?

Maybe could be add it an option to only hide the "All Users" link at top of left sidebar.

Of this way you only see the role's members and the same time could see items from other users, i think that the important is that users from one role can't see details from users alien to role.

**Reply by rijkvanzanten** [reply]
I think we'll be adding a dedicated "module allowlist" section to the roles page that allows you to enable/disable what modules are available per role 👍🏻

**Comment by gnimmelf** [comment]
In relation to this and the linked issue; 

Seeing that there is the same set of restrictions that also applies to *files* and *folders*, would it be an idea to add a system permissions preset for this?

Which also brings up the idea of expanding on system presets by *cascading* them: could they be presented in the right sidebar on role permissions, orderable multi-select + apply?

- Could modules for the role also be toggled somewhere on the page in addition to adding new modules? As of now, adding a new module will remove all the default modules.

Cheers,
Flemming

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
